lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -259,6 +260,158 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                       binder.op, resultType, operand);
                   return success();
                 });
+  patterns.onOp(
+      "Loop", 13, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        // Get all operands (maxTripCount, cond, ....inits....)
+        llvm::SmallVector<Value> operands;
+        if (binder.tensorOperandsList(operands) || operands.size() == 0) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Failed to get operands");
+        }
+
+        llvm::SmallVector<mlir::Type> operandTypeVec;
+        if (binder.tensorOperandTypes(operandTypeVec) ||
+            operandTypeVec.size() == 0) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Failed to get operandTypes");
+        }
+
+        Region *loopBodyIn;
+        if (binder.getRegionAtIndex(loopBodyIn, 0)) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Failed getting LoopBody Region");
+        }
+
+        // MaxTripCount - tensor int64 scalar (or empty)
+        Value maxTripCountTensor = operands[0];
+        auto maxTripCountInt = rewriter.create<Torch::AtenItemOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            maxTripCountTensor);
+
+        // Condition - tensor bool scalar (or empty)
+        Value conditionTensor = operands[1];
```

Comment:
Before doing this, you should check if this value exists or not. Since, the condition above only make sure that there should be > 0 operands, this operand could still be missing.

Reply:
> Actually onnx.Loop does have this constraint on the number of inputs
> 
> > Inputs
> > Between 2 and 2147483647 inputs.
> 
> But one can't be too careful these days, adding an extra check :)

Yes, there's a constraint on the number of inputs but that's for the `v_initial` arg, not the `cond` arg (it's optional and can be absent) which you're defining here: `Value conditionTensor = operands[1]`.  My comment is for this specific arg.